### DISCUSSION TITLE
Safe to use SyncScopedSession & AsyncSession in the same application

### DISCUSSION THREAD

**Discussion by erikwrede** [sqlalchemy/sqlalchemy#8613] [Usage Questions] [answered]
We are currently transitioning some applications to SQLAlchemy 1.4-style querying with `AsyncSession`. Are there any side effects or caveats to watch out for when switching from the traditional ScopedSession-Pattern to `sessionmaker(class_=AsyncSession)`?
Is it safe to use both in the same application at the same time? I checked the docs and could not find an answer to my question.

Thanks in advance!

**Comment by CaselIT** [comment, marked as answer]
Hi,

If you are referring to the `orm.scoped_session` then in general you cannot mix sync and async io in the same application.
It you are referring to the `async_scoped_session` then it should be fine to use it alongside the async session

**Reply by erikwrede** [reply]
Thanks for the quick reply! Does this restriction also apply to sessions on completely separate engines?

**Reply by CaselIT** [reply]
I meant as a way of programming. In general you should not use a sync engine from an async caller and vice versa.

If you have part of the application that's async and that uses an async engine and session (like an api server) and another part that uses a sync engine and session (like an background worker) then that should be fine.

Note that this is the same for any other network workflow, like you would not be using the `requests` library to make network calls from an async program
